Detectors/TPC/calibration/include/TPCCalibration/CalibLaserTracks.h

```diff
@@ -48,25 +82,39 @@ class CalibLaserTracks
                                                     mBz{other.mBz},
                                                     mDriftV{other.mDriftV},
                                                     mZbinWidth{other.mZbinWidth},
-                                                    mTFtime{other.mTFtime},
-                                                    mDVall{other.mDVall},
-                                                    mZmatchPairsTF{other.mZmatchPairsTF},
-                                                    mZmatchPairs{other.mZmatchPairs},
-                                                    mDVperTF{other.mDVperTF},
+                                                    mTFstart{other.mTFstart},
+                                                    mTFend{other.mTFend},
+                                                    mCalibDataTF{other.mCalibDataTF},
+                                                    mCalibData{other.mCalibData},
+                                                    mZmatchPairsTFA{other.mZmatchPairsTFA},
+                                                    mZmatchPairsTFC{other.mZmatchPairsTFC},
+                                                    mZmatchPairsA{other.mZmatchPairsA},
+                                                    mZmatchPairsC{other.mZmatchPairsC},
```

Comment:
Copy constructor resets `mFinalized`?

Reply:
Thanks, fixed.

Reply:
There are problems with `CalibLaserTracks(const CalibLaserTracks&) = default;` because of a CUDA compiler or something like this?

Reply:
The problem I get with default copy constructor is:
```
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/11.1.0/../../../../include/c++/11.1.0/memory:76:          
/usr/lib/gcc/x86_64-pc-linux-gnu/11.1.0/../../../../include/c++/11.1.0/bits/unique_ptr.h:962:34: error: call to implicitly-deleted copy constructor of 'o2::tpc::CalibLaserTracks'
    { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }                                                                                                                                                                                                                                                      
                                 ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                              
alicesw/sw/SOURCES/O2/O2-work/0/Detectors/Calibration/include/DetectorsCalibration/TimeSlot.h:40:25: note: in instantiation of function template specialization 'std::make_unique<o2::tpc::CalibLaserTracks, const o2::tpc::CalibLaserTracks &>' requested here                                       
      mContainer = std::make_unique<Container>(*src.getContainer());                                                                                                                                                                                                                                                          
                        ^                                                                                                                                                                                                                                                                                                     /usr/lib/gcc/x86_64-pc-linux-gnu/11.1.0/../../../../include/c++/11.1.0/bits/stl_algobase.h:719:18: note: in instantiation of member function 'o2::calibration::TimeSlot<o2::tpc::CalibLaserTracks>::operator=' requested here                                                   
            *--__result = std::move(*--__last);                                                                                                                                                                                                                                                                               
                        ^                                                                                                                                                                                                                                                                                                     
/usr/lib/gcc/x86_64-pc-linux-gnu/11.1.0/../../../../include/c++/11.1.0/bits/stl_algobase.h:760:24: note: in instantiation of function template specialization 'std::__copy_move_backward<true, false, std::random_access_iterator_tag>::__copy_move_b<o2::calibration::TimeSlot<o2::tpc::CalibLaserTracks> *, o2::calibration:
:TimeSlot<o2::tpc::CalibLaserTracks> *>' requested here                                                                                                        
                                       _Category>::__copy_move_b(__first,                                                                                                                                                                                                                                                     
...
...
...
```

Reply:
Ok. You have `std::unique_ptr<o2::utils::TreeStreamRedirector> mDebugStream;`. Which you ignore in the copy-constructor.

Reply:
Ahh right. Forgot about that. Yes, I wanted the keep the `TreeStreamRedirector` as `unique_ptr`, since it is only needed in debug mode.